### PR TITLE
RAB-928: Launch valid scheduled jobs

### PR DESCRIPTION
**Pull request by grena**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by ghoussard — APPROVED** [review]

**Review by grena — COMMENTED** [review]
Can't GTM because I'm the author of the PR, but it's GTM to me :+1:

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Tool/Component/BatchQueue/Queue/PublishJobToQueue.php [hunk 1]
```diff
@@ -160,23 +115,22 @@ private function createJobParameters(JobInterface $job, JobInstance $jobInstance
...
-                sprintf(
-                    'Job instance "%s" running the job "%s" with parameters "%s" is invalid because of "%s"',
-                    $code,
-                    $job->getName(),
-                    print_r($jobParameters->all(), true),
-                    $this->getErrorMessages($errors)
-                )
-            );
+
+        $jobInstanceViolations = $this->validator->validate($jobInstance, new JobInstanceConstraint());
```
Comment (StevenVAIDIE) [inline src/Akeneo/Tool/Component/BatchQueue/Queue/PublishJobToQueue.php:124]: You didn't need to give the JobInstanceConstraint, it's better to validate the whole job
```suggestion
        $jobInstanceViolations = $this->validator->validate($jobInstance);
```

File: src/Akeneo/Tool/Component/BatchQueue/spec/Queue/PublishJobToQueueSpec.php [hunk 2]
```diff
@@ -87,6 +88,8 @@ function it_publishes_a_job_to_the_execution_queue(
         $jobParametersFactory->create($job, [])->willReturn($jobParameters);
 
         $entityManager->merge($jobInstance)->shouldBeCalled();
+        $validator->validate($jobInstance, Argument::any())
```
Comment (StevenVAIDIE) [inline src/Akeneo/Tool/Component/BatchQueue/spec/Queue/PublishJobToQueueSpec.php:91]:
```suggestion
        $validator->validate($jobInstance)
```

File: src/Akeneo/Tool/Bundle/BatchBundle/Resources/config/validation/jobinstance.yml [hunk 3]
```diff
@@ -18,3 +18,5 @@ Akeneo\Tool\Component\Batch\Model\JobInstance:
             - Length:
                 max: 255
                 maxMessage: pim_import_export.validation.max_length_reached
+        automation:
```
Comment (StevenVAIDIE) [inline src/Akeneo/Tool/Bundle/BatchBundle/Resources/config/validation/jobinstance.yml:21]: Missing scheduled here

**Review by StevenVAIDIE — APPROVED** [review]

File: src/Akeneo/Tool/Component/Batch/spec/Updater/JobInstanceUpdaterSpec.php [hunk 4]
```diff
@@ -56,13 +56,83 @@ function it_updates_an_job_instance(
...
+        ]);
+    }
+
+    public function it_does_not_update_automation_setup_date_when_cron_expression_is_not_updated(
+        JobInstance $jobInstance,
+        ClockInterface $clock,
+    ): void
+    {
+        $currentAutomation = [
+            'cron_expression' => '0 0/4 * * *',
```
Comment (ValentinMumble) [inline src/Akeneo/Tool/Component/Batch/spec/Updater/JobInstanceUpdaterSpec.php:98]:
```suggestion
            'cron_expression' => '0 */4 * * *',
```

**Pull request merged**